src/python/gudhi/sklearn/rips_persistence.py

```diff
@@ -0,0 +1,148 @@
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Vincent Rouvreau
+#
+# Copyright (C) 2022 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+from .. import RipsComplex
+from sklearn.base import BaseEstimator, TransformerMixin
+
+import numpy as np
+# joblib is required by scikit-learn
+from joblib import Parallel, delayed
+
+# Mermaid sequence diagram - https://mermaid-js.github.io/mermaid-live-editor/
+# sequenceDiagram
+#     USER->>RipsPersistence: fit_transform(X)
+#     Note right of RipsPersistence: homology_dimensions=[i,j]
+#     RipsPersistence->>thread1: _tranform(X[0])
+#     RipsPersistence->>thread2: _tranform(X[1])
+#     Note right of RipsPersistence: ...
+#     thread1->>RipsPersistence: [array( Hi(X[0]) ), array( Hj(X[0]) )]
+#     thread2->>RipsPersistence: [array( Hi(X[1]) ), array( Hj(X[1]) )]
+#     Note right of RipsPersistence: ...
+#     RipsPersistence->>USER: [[array( Hi(X[0]) ), array( Hj(X[0]) )],<br/> [array( Hi(X[1]) ), array( Hj(X[1]) )],<br/> ...]
+
+
+class RipsPersistence(BaseEstimator, TransformerMixin):
+    """
+    This is a class for computing the persistence diagrams from a Vietoris-Rips complex.
+    """
+
+    def __init__(
+        self,
+        homology_dimensions,
+        max_edge_length=float('inf'),
+        input_type='point cloud',
+        nb_collapse=-1,
+        homology_coeff_field=11,
+        min_persistence=0.0,
+        expand_extra_dimension = True,
+        n_jobs=None,
+    ):
+        """
+        Constructor for the RipsPersistence class.
+
+        Parameters:
+            homology_dimensions (int or list of int): The returned persistence diagrams dimension(s).
+                Short circuit the use of :class:`~gudhi.representations.preprocessing.DimensionSelector` when only one
+                dimension matters (in other words, when `homology_dimensions` is an int).
+            max_edge_length (float): Rips value. Default is +Inf.
+            input_type (str): Can be 'point cloud' when inputs are point clouds, or 'lower distance matrix', when
+                inputs are lower triangular distance matrix (can be full square, but the upper part of the distance
+                matrix will not be considered). Default is 'point cloud'.
+            nb_collapse (int): The number of :func:`~gudhi.SimplexTree.collapse_edges` iterations to perform on the
+                SimplexTree. Default is -1, which means "automatic" (a relatively good enough number of iterations is
+                choosen).
+            homology_coeff_field (int): The homology coefficient field. Must be a prime number. Default value is 11.
+            min_persistence (float): The minimum persistence value to take into account (strictly greater than
+                `min_persistence`). Default value is `0.0`. Set `min_persistence` to `-1.0` to see all values.
```

Comment:
I wonder if we really want this option. It can make sense, for instance if you sample with a known precision. But in some cases we may be more interested in a minimal ratio death/birth (log-scale). If we don't have this parameter, it is easy to filter afterwards to get the same result. And filtering afterwards is likely not much slower than using the parameter min_persistence.
The reason I am thinking about this option is that supporting the case `min_persistence<0` is annoying. Edge-collapse removes a lot of intervals of length 0, so it makes little sense to have `min_persistence<0` and `nb_collapse>0`. Ripser also skips a lot of those intervals. So we would basically have to keep a slow path that does no edge-collapse and does not use ripser just for the not-so-useful case `min_persistence<0`.
Maybe a compromise could be to keep `min_persistence` but only support ≥0 values, or at least not promise much when it is <0.

Reply:
Oh, yes you are right that this can be easily done in a [preprocessing like class](https://github.com/GUDHI/gudhi-devel/blob/master/src/python/gudhi/representations/preprocessing.py). I removed min_persistence argument on 07706df8f